invenio_records_resources/services/uow.py

```diff
@@ -189,7 +190,7 @@ class IndexRefreshOp(Operation):
 
     def __init__(self, index, **kwargs):
         """Initialize the index to be refreshed."""
-        self._index = index
+        self._index = index.clone(name=build_alias_name(index._name))
```

Comment:
I'm slightly concerned about the performance of cloning. However,
- tests did not have an increase of time (4m50s while the previous run was 4m51s)
- the func description says _Create a copy of the instance with another name or connection alias. Useful for creating multiple indices with shared configuration_. So I'd venture to say it's for this purpose and things like the underlying connection is not duplicated.